plans/init.pp

```diff
@@ -0,0 +1,98 @@
+# Reboots targets and waits for them to be available again.
+#
+# @param nodes Targets to reboot.
+# @param message Message to log with the reboot (for platforms that support it).
+# @param reboot_delay How long (in seconds) to wait before rebooting. Defaults to 0, reboot immediately.
+#        Note on most Linux platforms, this time can only be specified in minutes, so non-0 values will
+#        delay reboot by at least 1 minute.
+# @param disconnect_wait How long (in seconds) to wait before checking whether the server has rebooted. Defaults to 1.
+# @param reconnect_wait How long (in seconds) to attempt to reconnect before giving up. Defaults to 120.
+# @param retry_interval How long (in seconds) to wait between retries. Defaults to 1.
+plan reboot (
+  TargetSpec $nodes,
+  Optional[String] $message = undef,
+  Integer[0] $reboot_delay = 0,
+  Integer[0] $disconnect_wait = 1,
```

Comment:
One second is nowhere near long enough. Only the simplest server will take >1 sec to disconnect. Anything remotely complex will take a fair bit longer than that

Reply:
Updated to 10. But as noted in a later comment, disconnect_wait is really only guidance to when to start testing that a reboot has occurred, not a necessary setting.